### ISSUE TITLE
Some fonts appear doubled when printed as PDF

### ISSUE THREAD

**Issue by hchauvat** [google/fonts#2330]
Hi, 

I have a weird issue where one piece of text using a Google Font appears twice when printed as PDF. 

I created a simple html file for demo purposes: 
```html
<!DOCTYPE html>
<html lang="en">
    <style>
        @import url('https://fonts.googleapis.com/css?family=IBM+Plex+Serif&display=swap');
        #doubled {
            font-family: 'IBM Plex Serif', serif;
        }
        #not-doubled {
            font-family: Helvetica, sans-serif;
        }
    </style>
<head>
    <meta charset="UTF-8">
    <title>Font Printing issue</title>
</head>
<body>
    <h2 id="doubled">Title in 'IBM Plex Serif' from Google Fonts that will be doubled</h2>
    <h2 id="not-doubled">Title in 'Helvetica' that will not be doubled</h2>
</body>
</html>
```
Then I simply perform a 'print as pdf' (either using the browser dialog or the one from macOS)
- Chrome (79.0.3945.130): no issue
- Edge (79.0.309.68): no issue
- Safari (13.0.4 (15608.4.9.1.3): the font is doubled
- Firefox (72.0.2): the font is doubled

Here are some zoomed-in screenshots:

### Chrome ✅ : 
![Chrome](https://user-images.githubusercontent.com/59441901/73761924-c927eb80-476f-11ea-95f6-f4d11c099457.png)
### Edge ✅ : 
![Edge](https://user-images.githubusercontent.com/59441901/73761925-c927eb80-476f-11ea-85a7-a82f37521f9f.png)
### Safari ❌ : 
![Safari](https://user-images.githubusercontent.com/59441901/73761929-c9c08200-476f-11ea-8ca3-f9115d9cba44.png)
### Firefox ❌ : 
![Firefox](https://user-images.githubusercontent.com/59441901/73761928-c9c08200-476f-11ea-9031-9db533d7e925.png)

You can see that the word `doubled` is made of 2 words stacked. 
It feels like this could be a browser issue maybe?

Thanks a lot for any help on this.

**Comment by laerm0** [comment]
Yep, confirmed, and like yours, Safari is not as bad as Firefox. These two browsers use different rendering engines. Windows Firefox produces a good PDF, but that's likely due to Windows' PDF print engine doing something different than OS X. I can't see any weirdness in the Firefox debugger, so I'm stumped. Maybe a bug in delivery to the browser? @rsheeter any thoughts?

**Comment by sladen** [comment]
@hchauvat, please could you attach the four resulting PDF files; this will allow comparison of the low-level PDF contents—in particular the subsetted versions of the fonts embedded into the PDFs.

**Comment by hchauvat** [comment]
Guys, 

You'll find enclosed the PDF files for each browser.

[Chrome.pdf](https://github.com/google/fonts/files/4203556/Chrome.pdf)
[Edge.pdf](https://github.com/google/fonts/files/4203557/Edge.pdf)
[Safari.pdf](https://github.com/google/fonts/files/4203554/Safari.pdf)
[Firefox.pdf](https://github.com/google/fonts/files/4203555/Firefox.pdf)

Thanks a lot!

**Comment by sladen** [comment]
```
$ for i in *.pdf ; do echo -n "$i: "; pdftotext -eol unix -raw $i - ; done | grep pdf
Chrome.pdf: Title in 'IBM Plex Serif' from Google Fonts that will be doubled
Edge.pdf: Title in 'IBM Plex Serif' from Google Fonts that will be doubled
Firefox.pdf: TTiittllee iinn ''IIBBMM PPlleexx SSeerriiff'' ffrroomm GGooooggllee FFoonnttss tthhaatt wwiillll bbee
Safari.pdf: Title in 'IBM Plex Serif' from Google Fonts that will beTitle in 'IBM Plex Serif' from Google Fonts that will be
```
With `pdftk x.pdf output x-uncompressed.pdf` we can observe that **Firefox** is setting a negative tracking equivalent to the width of the letter `T`, then manually stepping forewards for each character and drawing it twice:

```postscript
BT
  -0.5964 Tc
  [ (TT) -570 (i) -337 (i) -570 (t) -299 (t) -570 (l) -352 (l) -570 (e) -116 (e) ] TJ
ET
```

Whereas **Safari** is drawing the same string twice, with a slightly different matrix (`8 45`, then `9 45`)

```postscript
BT
  24 0 0 -24 8 45 Tm
  [ (Title) ] TJ
ET
BT
  24 0 0 -24 9 45 Tm
  [ (Title) ] TJ
ET
```

**Comment by hchauvat** [comment]
Nice in-depth analysis! 

Would you say that the issue comes from the browsers themselves? How come it doesn't appear with local/system fonts?

Thanks again!

**Comment by sladen** [comment]
@hchauvat please upload a PDF example "with local/system fonts", so that it is possible to have a look inside the PDF.

**Comment by hchauvat** [comment]
@sladen I believe the PDFs I have uploaded contain a sentence in Helvetica which is a system font on macOS. Don't know it that would do it.

**Comment by sladen** [comment]
@hchauvat to do a comparison requires 2+ things, that are nominally the same—the example of the  bug with is with IBM Plex; so a comparison "with local/system fonts" needs to also use IBM Plex to validate/isolate the claim.

**Comment by hchauvat** [comment]
@sladen You'll find enclosed some new PDFs. 

## The first batch (with a `-No_Swap` suffix): 
Without the `&display=swap;`parameter.
```css
@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Serif');

#doubled {
    font-family: 'IBM Plex Serif', serif;
}

#not-doubled {
    font-family: Helvetica, sans-serif;
}
```
```html
<h2 id="doubled">Title in 'IBM Plex Serif' from Google Fonts that will be doubled</h2>
<h2 id="not-doubled">Title in 'Helvetica' that will not be doubled</h2>
```
[Chrome-No_Swap.pdf](https://github.com/google/fonts/files/4204855/Chrome-No_Swap.pdf)
[Safari-No_Swap.pdf](https://github.com/google/fonts/files/4204853/Safari-No_Swap.pdf)
[Firefox-No_Swap.pdf](https://github.com/google/fonts/files/4204854/Firefox-No_Swap.pdf)

## The second batch (with a `-Local_vs_Remote` suffix):
Using both a GFont ('remote') and a local version of the 'IBM Plex Serif' font
```css
@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Serif&display=swap');

#remote {
    font-family: 'IBM Plex Serif', serif;
}

@font-face {
    font-family: 'IBMPlexSerif-Local';
    font-style: normal;
    font-weight: 400;
    src: url('./font/IBMPlexSerif-Regular-Local.ttf');
}

#local {
    font-family: 'IBMPlexSerif-Local', serif;
}

#not-doubled {
    font-family: Helvetica, sans-serif;
}
```
```html
<h2 id="remote">Title in 'IBM Plex Serif' as a remote GFont</h2>
<h2 id="local">Title in 'IBM Plex Serif' as a local font</h2>
<h2 id="not-doubled">Title in 'Helvetica' that will not be doubled</h2>
```
[Chrome-Local_vs_Remote.pdf](https://github.com/google/fonts/files/4204860/Chrome-Local_vs_Remote.pdf)
[Firefox-Local_vs_Remote.pdf](https://github.com/google/fonts/files/4204851/Firefox-Local_vs_Remote.pdf)
[Safari-Local_vs_Remote.pdf](https://github.com/google/fonts/files/4204852/Safari-Local_vs_Remote.pdf)

**Comment by laerm0** [comment]
(thanks for the hand @sladen 👍 )

**Comment by hchauvat** [comment]
A quick update, 

The issue doesn't appear if the text is in a `<p>` element, only in a `<hx>` tag.

**Comment by sladen** [comment]
So… this probably

1.`h1`…`h6` default to bold, 
2. no `font-weight: bold;` having been supplied
3. both browsers providing a faux "bold" by synthesis—using the regular weight twice, slightly offset

For Helvetica, as a standard Postscript/PDF font, it will be full mapped/available locally, including bold.  Things to try:

```
h1,h2,h3,h4,h5,h6 { font-weight: normal; }
```

or:

```
@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Serif:400,700')
```

**Comment by hchauvat** [comment]
Both your suggestions work. The 1st one, everywhere (both remote and local fonts). The 2nd fixes only the remote one (obviously).

Nice catch!

**Comment by curiousercreative** [comment]
Can this be closed? Seems to not be a bug with Google Fonts.
